lib/writer/block-write-stream.js

```diff
@@ -134,10 +140,26 @@ class BlockWriteStream extends stream.Writable {
     }
 
     fs.write(this.fd, chunk, 0, chunk.length, chunk.position, (error, bytesWritten) => {
-      this.bytesWritten += bytesWritten
-      this.delta += bytesWritten
-      this.blocksWritten += 1
-      this.position += bytesWritten
+      if (!error) {
+        this.bytesWritten += bytesWritten
+        this.delta += bytesWritten
+        this.blocksWritten += 1
+        this.position += bytesWritten
+        this.retries = 0
+        next(error)
+        return
+      }
+
+      const isTransient = errors.isTransientError(error)
+
+      if (isTransient && (this.retries < this.maxRetries)) {
+        this.retries += 1
+        this._write(chunk, encoding, next)
+        return
+      } else if (isTransient) {
```

Comment:
Minor nit, but this looks a bit weird. Could you instead do:
```
if (isTransient) {
    if (this.retries < this.maxRetries) {
    } else {
    }
}
```
?

Reply:
Excessive branching ain't good for perf; better to keep conditions as flat as possible

Reply:
Huh? I wouldn't expert there to be much performance difference (if any) between that and the:
```
if (isTransient && (this.retries < this.maxRetries)) {
    // stuff
} else if (isTransient) {
    // other stuff
}
```
that you're using currently? Note that in both versions there's two `if` statements and one `else` statement, they're just arranged slightly differently ;-)

But then I obviously have no idea how NodeJS's internal optimisations work, so perhaps there really is a difference?